static/app/views/issueDetails/streamline/issueCronCheckTimeline.tsx

```diff
@@ -0,0 +1,243 @@
+      enabled: !hasCronDetector,
+      retry: false,
+    }
+  );
+
+  // Fall back to the fetched event since the legacy UI isn't nested within the provider the provider
+  return hasCronDetector
+    ? detectorId
+    : event?.tags?.find(({key}) => key === 'monitor.id')?.value;
+}
+
+function useCronLegendStatuses({bucketStats}: {bucketStats: MonitorBucket[]}) {
+  /**
+   * Extract a list of statuses that have occurred at least once in the bucket stats.
+   */
+  return useMemo(() => {
+    const statusMap: Record<CheckInStatus, boolean> = {
+      [CheckInStatus.OK]: true,
+      [CheckInStatus.ERROR]: false,
+      [CheckInStatus.IN_PROGRESS]: false,
+      [CheckInStatus.MISSED]: false,
+      [CheckInStatus.TIMEOUT]: false,
+      [CheckInStatus.UNKNOWN]: false,
+    };
+    bucketStats?.forEach(([_timestamp, bucketEnvMapping]) => {
+      const bucketEnvMappingEntries = Object.values(bucketEnvMapping) as Array<
+        StatsBucket<CheckInStatus>
+      >;
+      for (const statBucket of bucketEnvMappingEntries) {
+        const statBucketEntries = Object.entries(statBucket) as Array<
+          [CheckInStatus, number]
+        >;
+        for (const [status, count] of statBucketEntries) {
+          if (count > 0 && !statusMap[status]) {
+            statusMap[status] = true;
+          }
+        }
+      }
+    });
+    return (Object.keys(statusMap) as CheckInStatus[]).filter(
+      status => statusMap[status]
+    );
+  }, [bucketStats]);
+}
+
+export function IssueCronCheckTimeline({group}: {group: Group; event?: Event}) {
+  const elementRef = useRef<HTMLDivElement>(null);
+  const {width: containerWidth} = useDimensions<HTMLDivElement>({elementRef});
+  const timelineWidth = useDebouncedValue(containerWidth, 500);
+  const timeWindowConfig = useTimeWindowConfig({timelineWidth});
+
+  const cronAlertId = useCronIssueAlertId({groupId: group.id});
+  const {data: stats, isPending} = useMonitorStats({
+    monitors: cronAlertId ? [cronAlertId] : [],
+    timeWindowConfig,
+  });
+
+  const cronStats = useMemo(() => {
+    if (!cronAlertId) {
+      return [];
+    }
+    return stats?.[cronAlertId] ?? [];
+  }, [cronAlertId, stats]);
+
+  const statEnvironments = useMemo(() => {
+    const envSet = cronStats.reduce((acc, [_, envs]) => {
+      return new Set([...acc, ...Object.keys(envs)]);
+    }, new Set<string>());
+    return [...(envSet ?? [])];
+  }, [cronStats]);
+
+  const legendStatuses = useCronLegendStatuses({
+    bucketStats: cronStats,
+  });
+
+  return (
+    <ChartContainer envCount={statEnvironments.length}>
+      <TimelineLegend ref={elementRef}>
+        {!isPending &&
+          legendStatuses.map(status => (
+            <Flex align="center" gap={space(0.5)} key={status}>
+              <MonitorIndicator status={status as CheckInStatus} size={8} />
+              <TimelineLegendText>
+                {statusToText[status as CheckInStatus]}
+              </TimelineLegendText>
+            </Flex>
```

Comment:
i think status is already CheckInStatus
```suggestion
            <Flex align="center" gap={space(0.5)} key={status}>
              <MonitorIndicator status={status} size={8} />
              <TimelineLegendText>{statusToText[status]}</TimelineLegendText>
            </Flex>
```